chromium_src/chrome/browser/net/stub_resolver_config_reader.cc

```diff
@@ -0,0 +1,138 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "chrome/browser/net/stub_resolver_config_reader.h"
+#include "base/feature_list.h"
+#include "brave/components/brave_vpn/buildflags/buildflags.h"
+#include "build/buildflag.h"
+#include "chrome/browser/net/secure_dns_config.h"
+#include "chrome/common/pref_names.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/browser/network_service_instance.h"
+#include "net/dns/public/dns_over_https_config.h"
+#include "services/network/public/mojom/network_service.mojom.h"
+#include "third_party/abseil-cpp/absl/types/optional.h"
+
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
+#include "brave/components/brave_vpn/features.h"
+#endif
+
+#if BUILDFLAG(IS_WIN)
+namespace {
+
+bool ShouldOverride(net::SecureDnsMode secure_dns_mode,
+                    PrefService* local_state,
+                    SecureDnsConfig::ManagementMode management_mode,
+                    bool is_managed) {
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
+  if (!base::FeatureList::IsEnabled(
+          brave_vpn::features::kBraveVPNDnsProtection)) {
+    return false;
+  }
+  if (secure_dns_mode == net::SecureDnsMode::kSecure)
+    return false;
```

Comment:
add comment `No need to override if the user already has already configured doh`

Reply:
done here https://github.com/brave/brave-core/pull/15989/commits/42dccf020f860c9521be7bac9f6d98670b6e9d68

Reply:
needs to be re-added